### ISSUE TITLE
[History Error] (Version: stellar-core 12.2.0)

### ISSUE THREAD

**Issue by alexqrid** [stellar/stellar-core#2372]
I've had synced node running without any problem for several months. But suddenly it stopped to sync new blocks and show me the following error message:
```
2019-12-31T19:02:49.281 GDRU2 [History ERROR] Bad ledger-header history entry: claimed ledger [seq=27533706, hash=450c7b] does not agree with LCL [seq=27533706, hash=9f1266] [VerifyLedgerChainWork.cpp:160]
2019-12-31T19:02:49.281 GDRU2 [History ERROR] Catchup material failed verification - hash mismatch, propagating failure [VerifyLedgerChainWork.cpp:337]
2019-12-31T19:02:49.281 GDRU2 [History ERROR] One or more of history archives may be corrupted. Update HISTORY configuration entry to only contain valid ones [VerifyLedgerChainWork.cpp:339]
2019-12-31T19:02:51.113 GDRU2 [History ERROR] Bad ledger-header history entry: claimed ledger [seq=27533706, hash=450c7b] does not agree with LCL [seq=27533706, hash=9f1266] [VerifyLedgerChainWork.cpp:160]
2019-12-31T19:02:51.113 GDRU2 [History ERROR] Catchup material failed verification - hash mismatch, propagating failure [VerifyLedgerChainWork.cpp:337]
2019-12-31T19:02:51.113 GDRU2 [History ERROR] One or more of history archives may be corrupted. Update HISTORY configuration entry to only contain valid ones [VerifyLedgerChainWork.cpp:339]
2019-12-31T19:02:52.929 GDRU2 [History ERROR] Bad ledger-header history entry: claimed ledger [seq=27533706, hash=450c7b] does not agree with LCL [seq=27533706, hash=9f1266] [VerifyLedgerChainWork.cpp:160]
2019-12-31T19:02:52.929 GDRU2 [History ERROR] Catchup material failed verification - hash mismatch, propagating failure [VerifyLedgerChainWork.cpp:337]
2019-12-31T19:02:52.930 GDRU2 [History ERROR] One or more of history archives may be corrupted. Update HISTORY configuration entry to only contain valid ones [VerifyLedgerChainWork.cpp:339]
2019-12-31T19:02:56.265 GDRU2 [History ERROR] Bad ledger-header history entry: claimed ledger [seq=27533706, hash=450c7b] does not agree with LCL [seq=27533706, hash=9f1266] [VerifyLedgerChainWork.cpp:160]
2019-12-31T19:02:56.265 GDRU2 [History ERROR] Catchup material failed verification - hash mismatch, propagating failure [VerifyLedgerChainWork.cpp:337]
2019-12-31T19:02:56.265 GDRU2 [History ERROR] One or more of history archives may be corrupted. Update HISTORY configuration entry to only contain valid ones [VerifyLedgerChainWork.cpp:339]
2019-12-31T19:03:11.609 GDRU2 [History ERROR] Bad ledger-header history entry: claimed ledger [seq=27533706, hash=450c7b] does not agree with LCL [seq=27533706, hash=9f1266] [VerifyLedgerChainWork.cpp:160]
2019-12-31T19:03:11.610 GDRU2 [History ERROR] Catchup material failed verification - hash mismatch, propagating failure [VerifyLedgerChainWork.cpp:337]
2019-12-31T19:03:11.610 GDRU2 [History ERROR] One or more of history archives may be corrupted. Update HISTORY configuration entry to only contain valid ones [VerifyLedgerChainWork.cpp:339]
2019-12-31T19:03:14.964 GDRU2 [History ERROR] Bad ledger-header history entry: claimed ledger [seq=27533706, hash=450c7b] does not agree with LCL [seq=27533706, hash=9f1266] [VerifyLedgerChainWork.cpp:160]
2019-12-31T19:03:14.964 GDRU2 [History ERROR] Catchup material failed verification - hash mismatch, propagating failure [VerifyLedgerChainWork.cpp:337]
2019-12-31T19:03:14.964 GDRU2 [History ERROR] One or more of history archives may be corrupted. Update HISTORY configuration entry to only contain valid ones [VerifyLedgerChainWork.cpp:339]
2019-12-31T19:03:14.964 GDRU2 [Ledger ERROR] Catchup will restart at next close. [LedgerManagerImpl.cpp:691]
```  
After restarting node several times but I got the same error everytime:  
<details> <summary>Error</summary>  

```
2019-12-31T18:28:13.208 GDQMO [History ERROR] Catchup material failed verification - hash mismatch, propagating failure [VerifyLedgerChainWork.cpp:337]
2019-12-31T18:28:13.208 GDQMO [History ERROR] One or more of history archives may be corrupted. Update HISTORY configuration entry to only contain valid ones [VerifyLedgerChainWork.cpp:339]
2019-12-31T18:28:18.452 GDQMO [History ERROR] Bad ledger-header history entry: claimed ledger [seq=27533706, hash=450c7b] does not agree with LCL [seq=27533706, hash=9f1266] [VerifyLedgerChainWork.cpp:160]
2019-12-31T18:28:18.452 GDQMO [History ERROR] Catchup material failed verification - hash mismatch, propagating failure [VerifyLedgerChainWork.cpp:337]
2019-12-31T18:28:18.453 GDQMO [History ERROR] One or more of history archives may be corrupted. Update HISTORY configuration entry to only contain valid ones [VerifyLedgerChainWork.cpp:339]
2019-12-31T18:28:19.709 GDQMO [History ERROR] Bad ledger-header history entry: claimed ledger [seq=27533706, hash=450c7b] does not agree with LCL [seq=27533706, hash=9f1266] [VerifyLedgerChainWork.cpp:160]
2019-12-31T18:28:19.709 GDQMO [History ERROR] Catchup material failed verification - hash mismatch, propagating failure [VerifyLedgerChainWork.cpp:337]
2019-12-31T18:28:19.709 GDQMO [History ERROR] One or more of history archives may be corrupted. Update HISTORY configuration entry to only contain valid ones [VerifyLedgerChainWork.cpp:339]
2019-12-31T18:28:33.943 GDQMO [History ERROR] Bad ledger-header history entry: claimed ledger [seq=27533706, hash=450c7b] does not agree with LCL [seq=27533706, hash=9f1266] [VerifyLedgerChainWork.cpp:160]
2019-12-31T18:28:33.944 GDQMO [History ERROR] Catchup material failed verification - hash mismatch, propagating failure [VerifyLedgerChainWork.cpp:337]
2019-12-31T18:28:33.944 GDQMO [History ERROR] One or more of history archives may be corrupted. Update HISTORY configuration entry to only contain valid ones [VerifyLedgerChainWork.cpp:339]
2019-12-31T18:28:33.944 GDQMO [Ledger ERROR] Catchup will restart at next close. [LedgerManagerImpl.cpp:691]

```
</details>

Could not find any solutions, so I initialized new DB with `new-db` command, and started to sync my node from scratch. 
Actually I could not realize what was the problem.
<details> <summary>My node's config </summary>  

```
HTTP_PORT=10000
PUBLIC_HTTP_PORT=false
COMMANDS=[
"ll?level=ERROR"
]
ENTRY_CACHE_SIZE=10
BEST_OFFERS_CACHE_SIZE=10
LOG_FILE_PATH=secret_path_to_log
BUCKET_DIR_PATH=secret_path_to_dir
NETWORK_PASSPHRASE="Public Global Stellar Network ; September 2015"
PEER_PORT=11625
TARGET_PEER_CONNECTIONS=5
MAX_ADDITIONAL_PEER_CONNECTIONS=-1
MAX_PENDING_CONNECTIONS=2
PEER_AUTHENTICATION_TIMEOUT=2
KNOWN_PEERS=[
"core-live-a.stellar.org",
"core-live-b.stellar.org",
"core-live-c.stellar.org",
"au.stellar.ibm.com",
"no.stellar.ibm.com",
"uk.stellar.ibm.com",
"ca.stellar.ibm.com",
]
NODE_IS_VALIDATOR=false
FAILURE_SAFETY=0
UNSAFE_QUORUM=true
CATCHUP_COMPLETE=false
CATCHUP_RECENT=20000
DATABASE= SECRET
MAX_CONCURRENT_SUBPROCESSES=2
AUTOMATIC_MAINTENANCE_PERIOD=14400
AUTOMATIC_MAINTENANCE_COUNT=0
NODE_NAMES=[
"GCKWUQGSVO45ZV3QK7POYL7HMFWDKWJVMFVEGUJKCAEVUITUCTQWFSM6  IBM_Australia",
"GB2HF2NHRKKFZYFDGD7MUENOYROOEK7SWYV2APYOODP6P7BUJTLILKIL  IBM_Canada",
"GDRA72H7JWXAXWJKOONQOPH3JKNSH5MQ6BO5K74C3X6FO2G3OG464BPU  IBM_Norway",
"GAENPO2XRTTMAJXDWM3E3GAALNLG4HVMKJ4QF525TR25RI42YPEDULOW  IBM_UK",
"GCGB2S2KGYARPVIA37HYZXVRM2YZUEXA6S33ZU5BUDC6THSB62LZSTYH  sdf1",
"GCM6QMP3DLRPTAZW2UZPCPX2LF3SXWXKPMP3GKFZBDSF3QZGV2G5QSTK  sdf2",
"GABMKJM6I25XI4K7U6XWMULOUQIQ27BCTMLS6BYYSOWKTBUXVRJSXHYQ  sdf3"]
KNOWN_CURSORS=["HORIZON"]
[QUORUM_SET]
THRESHOLD_PERCENT=66
VALIDATORS=[
"$sdf1", "$sdf2", "$sdf3"
]
[QUORUM_SET.1]
THRESHOLD_PERCENT=67
VALIDATORS=[
"$IBM_UK","$IBM_Australia","$IBM_Norway","$IBM_Canada"
]
[HISTORY.sdf1]
get="curl -sf http://history.stellar.org/prd/core-live/core_live_001/{0} -o {1}"
[HISTORY.sdf2]
get="curl -sf http://history.stellar.org/prd/core-live/core_live_002/{0} -o {1}"
[HISTORY.sdf3]
get="curl -sf http://history.stellar.org/prd/core-live/core_live_003/{0} -o {1}"
```
</details>

### Environment
*  Linux 4.15.0-72-generic # 81-Ubuntu SMP Tue Nov 26 12:20:02 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux
* Running from command line
* Did you pass in special parameters when building the app? No.

**Comment by MonsieurNicolas** [comment]
Can you fill the information from the bug template? Without version information on your system, it's hard to tell what the issue could be

**Comment by alexqrid** [comment]
The same problem occured again a few hours ago.
```
2020-01-07T20:50:28.948 GBXWP [History ERROR] Bad ledger-header history entry: claimed ledger [seq=27644519, hash=4366c2] does not agree with LCL [seq=27644519, hash=ca621b] [VerifyLedgerChainWork.cpp:160]
2020-01-07T20:50:28.949 GBXWP [History ERROR] Catchup material failed verification - hash mismatch, propagating failure [VerifyLedgerChainWork.cpp:337]
2020-01-07T20:50:28.949 GBXWP [History ERROR] One or more of history archives may be corrupted. Update HISTORY configuration entry to only contain valid ones [VerifyLedgerChainWork.cpp:339]
2020-01-07T20:50:44.147 GBXWP [History ERROR] Bad ledger-header history entry: claimed ledger [seq=27644519, hash=4366c2] does not agree with LCL [seq=27644519, hash=ca621b] [VerifyLedgerChainWork.cpp:160]
2020-01-07T20:50:44.147 GBXWP [History ERROR] Catchup material failed verification - hash mismatch, propagating failure [VerifyLedgerChainWork.cpp:337]
2020-01-07T20:50:44.147 GBXWP [History ERROR] One or more of history archives may be corrupted. Update HISTORY configuration entry to only contain valid ones [VerifyLedgerChainWork.cpp:339]
2020-01-07T20:50:44.147 GBXWP [Ledger ERROR] Catchup will restart at next close. [LedgerManagerImpl.cpp:691]
```

**Comment by marta-lokhova** [comment]
Hello!

It looks like your database somehow got into a corrupt state, since the ledger hash from history archive looks correct. A few questions to help us debug this. How did the node get out of sync and trigger "catchup"? Specifically, did it crash before you re-started it? Or did you run `new-db`, then started catching up, and then the error occurred?

**Comment by alexqrid** [comment]
> How did the node get out of sync and trigger "catchup"?

Actually I do not know why did it get out of sync.
> Specifically, did it crash before you re-started it?

No, my node was running for several months in synced state, but then that error occurred.

> did you run `new-db`, then started catching up, and then the error occurred?

I was able to fix this issue by running `new-db` and starting node to sync again, but after 15 hours the same error occurred. Then I increased `MAX_ADDITIONAL_PEER_CONNECTIONS`, and changed my quorums configuration, added a few nodes also several nodes with history archives and initialized `new-db` and `run` the node. But  after a couple of days the same happened again.

Last time the problem happened on `Jan 17 07:00:04 UTC 2020`. Then I executed `new-db` again and ran the node. So to my luck it's still running:
```
"startedOn" : "2020-01-17T10:13:31Z",
 "state" : "Synced!"
``` 
Because I did not change any valuable configurations that can influence the catchup and everything is working fine now, I suppose that the problem was with the network, I am using Digital Ocean's VPS and may be there were some problems with the network in the DC.
